bitcoin/src/address.rs

```diff
@@ -848,10 +848,18 @@ impl Address {
 	self.address_type_internal()
     }
 
+    /// Checks whether or not the address is following Bitcoin standardness rules when
+    /// *spending* from this address.
+    ///
+    /// SegWit addresses with unassigned witness versions or non-standard program sizes are
+    /// considered non-standard.
+    pub fn is_spend_standard(&self) -> bool { self.address_type().is_some() }
```

Comment:
If I'm correct that receiver statically knows the address is standard then I think the method is useless and we should just remove it. If there is any use then I want a warning in doc that this should **not** be called by the sender.

Reply:
> If I'm correct that receiver statically knows the address is standard

This method is preciously how the receiver would know that the address they are receiving the money to is standard.

Reply:
The receiver is the one who constructs the address so I would expect this to be handled by construction.

Reply:
Are you suggesting two structures? `SendingAddress` and `RecievingAddress`. Or perhaps yet another phantom data type in Address itself. We want to support sending to unknown addresses, but don't want to receive it ourselves. I don't see how we can handle this under the current `Address` construction. 

That being said, I don't see any utility for wallet developers who want to use this method. It maybe useful enthusiasts who scan the blockchain for non-standard spends or for block explorers. I recall somebody did a taproot spend before taproot was activated to fork off UASF nodes.

Reply:
> Are you suggesting two structures?

Not really. If I write an application that uses some of the common ways of receiving then I write the code such that `Address` is always valid and don't need to check it. Would would I even do if it's wrong?

> That being said, I don't see any utility for wallet developers who want to use this method.

Yeah, I believe we both don't for the same reason.

> It maybe useful enthusiasts who scan the blockchain for non-standard spends or for block explorers.

Shouldn't the scripts be checked instead of addresses?

> I recall somebody did a taproot spend before taproot was activated to fork off UASF nodes.

OT but sounds like that person was confused? I don't see how it could fork off UASF nodes.

Reply:
>Not really. If I write an application that uses some of the common ways of receiving then I write the code such that Address is always valid and don't need to check it.

This is why it's a utility method and not something that we're gating constructors on or enforcing in the type system.

>Would would I even do if it's wrong?

Lol assert probably. But I think it's most common to hit cases like this during transition periods...like, maybe your wallet would let you generate a Taproot address before Taproot was active, but would put a bunch of warnings on it or otherwise try to prevent you from using it until it became standard.

>Shouldn't the scripts be checked instead of addresses?

This is an interesting thought. Maybe users who want to do this should need to first convert the address to a scriptpubkey, and then do a standardness check on that. But I think such an API change should wait for script tagging.